doc/source/security/enabling-master-key.rst

```diff
@@ -0,0 +1,206 @@
+------------------------------
+
+:Availability: This feature is **Experimental**.
+
+To enable system tablespace encryption, edit the my.cnf file with the following:
+
+* Add ``innodb_sys_tablespace_encrypt``
+* Edit the ``innodb_sys_tablespace_encrypt`` value to "ON"
+
+System tablespace encryption can only be enabled with the ``--initialize``
+option. You cannot encrypt existing tables in the System tablespace.
+.. _encrypting-undo-log:
+
+Encrypting the Undo Log Data
+------------------------------
+
+:Availability: This feature is **Experimental**.
+
+You enable encryption for an undo log data by adding the following line to the
+my.cnf file:
+
+.. code-block:: MySQL
+
+  [mysqld]
+  innodb_undo_log_encrypt=ON
+
+.. _encrypting-binary-logs:
+
+Encrypting Binary Logs
+-----------------------
+
+:Availability: This feature is **Experimental** quality.
+
+Enable encryption of binary logs by running the following statement:
+
+.. code-block:: MySQL
+
+    binlog_encryption=ON
+
+You can rotate the encryption key used by |Percona Server| by running the
+following statement:
+
+.. code-block:: MySQL
+
+    mysql> SELECT rotate_system_key("percona_binlog");
+
+This command creates a new binlog encryption key in the keyring. The new key
+encrypts the next binlog file.
+
+.. _verifying-encryption:
+
+Verifying the Encryption Setting
+----------------------------------
+
+For single tablespaces, verify the ENCRYPTION option using
+`INFORMATION_SCHEMA.TABLES` and the `CREATE OPTIONS` settings.
+
+.. code-block:: MySQL
+
+    mysql> SELECT TABLE_SCHEMA, TABLE_NAME, CREATE_OPTIONS FROM
+           INFORMATION_SCHEMA.TABLES WHERE CREATE_OPTIONS LIKE '%ENCRYPTION%';
+
+    +----------------------+-------------------+------------------------------+
+    | TABLE_SCHEMA         | TABLE_NAME        | CREATE_OPTIONS               |
+    +----------------------+-------------------+------------------------------+
+    |sample                | t1                | ENCRYPTION="Y"               |
+    +----------------------+-------------------+------------------------------+
+
+A ``flag`` field in the ``INFORMATION_SCHEMA.INNODB_TABLESPACES`` has the bit
+number 13 set if the tablespace is encrypted. This bit can be checked with the
+``flag & 8192`` expression with the following method:
+
+.. code-block:: mysql
+
+    SELECT space, name, flag, (flag & 8192) != 0 AS encrypted FROM
+    INFORMATION_SCHEMA.INNODB_TABLESPACES WHERE name in ('foo', 'test/t2', 'bar',
+    'noencrypt');
+
+      +-------+-----------+-------+-----------+
+      | space | name      | flag  | encrypted |
+      +-------+-----------+-------+-----------+
+      |    29 | foo       | 10240 |      8192 |
+      |    30 | test/t2   |  8225 |      8192 |
+      |    31 | bar       | 10240 |      8192 |
+      |    32 | noencrypt |  2048 |         0 |
+      +-------+-----------+-------+-----------+
+      4 rows in set (0.01 sec)
+
+:Availability: This feature is **Experimental**.
```

Comment:
@patrickbirch @georgelorchpercona as it was decided that we are going to remove the encryption threads from 5.7 and INFORMATION_SCHEMA.INNODB_TABLESPACE_ENCRYPTION is part of the encryption threads WL - shouldn't the information on INNODB_TABLESPACE_ENCRYPTION be removed ? I think yes.

Reply:
agreed

Reply:
Removed the section about the I_S.Innodb_tablespace_encryption description.